### PR TITLE
docs: clarify Tour of Heroes, esp. Routing step

### LINKED ISSUE #21207: Tour Of Heroes > Missing Step > exports: [ RouterModule ]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Feature request
[x] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior
The current documentation within the Tour of Heros is missing a step I believe...

https://angular.io/tutorial/toh-pt5#routermoduleforroot

The step calls for adding...

`imports: [ RouterModule.forRoot(routes) ],`

which results in an error of...

```
compiler.js:485 Uncaught Error: Template parse errors:
'router-outlet' is not a known element:
1. If 'router-outlet' is an Angular component, then verify that it is part of this module.
2. If 'router-outlet' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message. ("<h1>{{ title }}</h1>
[ERROR ->]<router-outlet></router-outlet>
<app-messages></app-messages>
```

## Expected behavior

I am new to Angular, but I believe what is needed is an export as well. Or at least that is how I was able to get things working as follows...

```
  imports: [ RouterModule.forRoot(routes) ],
  exports: [ RouterModule ]
```

### PR DESCRIPTION
**Pull request by kapunahelewong**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number:
Closes #21207

## What is the new behavior?

Clarify instructions in toh, especially in the Routing step.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

Cleared up ambiguity in the Routing step.
Moved `httpOptions` inside of service class.
Add or fix prose to clarify instructions.
Added headers to code snippets.
Fixed CSS in examples by removing `!important`.

### CODE REVIEW

File: aio/content/examples/toh-pt6/src/app/hero.service.ts [hunk 1]
```diff
@@ -96,8 +97,8 @@ export class HeroService {
   // #docregion addHero
   /** POST: add a new hero to the server */
   addHero (hero: Hero): Observable<Hero> {
-    return this.http.post<Hero>(this.heroesUrl, hero, httpOptions).pipe(
-      tap((newHero: Hero) => this.log(`added hero w/ id=${newHero.id}`)),
+    return this.http.post<Hero>(this.heroesUrl, hero, this.httpOptions).pipe(
+      tap((hero: Hero) => this.log(`added hero w/ id=${hero.id}`)),
```
Comment (brandonroberts) [inline aio/content/examples/toh-pt6/src/app/hero.service.ts]: Keep this as `newHero` so we don't have two shadow `hero` variables.

File: aio/content/tutorial/toh-pt2.md [hunk 2]
```diff
@@ -21,19 +21,17 @@ header="src/app/mock-heroes.ts">
...
-
 Open the `HeroesComponent` class file and import the mock `HEROES`.
 
 <code-example path="toh-pt2/src/app/heroes/heroes.component.ts" region="import-heroes" header="src/app/heroes/heroes.component.ts (import HEROES)">
 </code-example>
 
 In the same file (`HeroesComponent` class), define a component property called `heroes` to expose `HEROES` array for binding.
 
-<code-example path="toh-pt2/src/app/heroes/heroes.component.ts" region="component">
+<code-example path="toh-pt2/src/app/heroes/heroes.component.ts" header="src/app/heroes/heroes.component.ts" region="component">
```
Comment (brandonroberts) [inline aio/content/tutorial/toh-pt2.md]: Include the closing curly brace for the class in this snippet

**Review by brandonroberts — DISMISSED** [review]

File: aio/content/examples/toh-pt6/src/app/hero.service.ts (hunk 1 above)
Comment (kapunahelewong) [reply aio/content/examples/toh-pt6/src/app/hero.service.ts]: got it.

File: aio/content/tutorial/toh-pt2.md (hunk 2 above)
Comment (kapunahelewong) [reply aio/content/tutorial/toh-pt2.md]: done!

File: aio/content/tutorial/toh-pt6.md [hunk 3]
```diff
@@ -11,174 +11,147 @@ When you're done with this page, the app should look like this <live-example></l
...
 
-The class `src/app/in-memory-data.service.ts` is generated by the following command:
-
-<code-example language="sh" class="code-shell">
-  ng generate service InMemoryData
+<code-example path="toh-pt6/src/app/app.module.ts" region="import-in-mem-stuff" header="src/app/app.module.ts (In-memory Web API imports)">
 </code-example>
 
-This class has the following content:
+After the `HttpClientModule` add the `HttpClientInMemoryWebApiModule`
```
Comment (jbogarthyde) [inline aio/content/tutorial/toh-pt6.md]: After the `HttpClientModule`, add
(add comma)

File: aio/content/tutorial/toh-pt6.md [hunk 4]
```diff
@@ -11,174 +11,147 @@ When you're done with this page, the app should look like this <live-example></l
...
 
-HTTP is a request/response protocol. 
+HTTP is a request/response protocol.
 You make a request, it returns a single response.
 
 In general, an observable _can_ return multiple values over time.
 An observable from `HttpClient` always emits a single value and then completes, never to emit again.
 
-This particular `HttpClient.get` call returns an `Observable<Hero[]>`, literally "_an observable of hero arrays_". In practice, it will only return a single hero array.
+This particular `HttpClient.get()` call returns an `Observable<Hero[]>`, that is, "_an observable of hero arrays_". In practice, it will only return a single hero array.
```
Comment (jbogarthyde) [inline aio/content/tutorial/toh-pt6.md]: an `Observable<Hero[]>`; that is, ...
(change comma to semicolon)

File: aio/content/tutorial/toh-pt6.md [hunk 5]
```diff
@@ -413,143 +369,125 @@ In this last exercise, you learn to chain `Observable` operators together
...
 
 <code-example language="sh" class="code-shell">
   ng generate component hero-search
 </code-example>
 
-The CLI generates the three `HeroSearchComponent` files and adds the component to the `AppModule` declarations
+The CLI generates the three `HeroSearchComponent` files and adds the component to the `AppModule` declarations.
 
-Replace the generated `HeroSearchComponent` _template_ with a text box and a list of matching search results like this.
+Replace the generated `HeroSearchComponent` template with an `<input>` and a list of matching search results with the following.
```
Comment (jbogarthyde) [inline aio/content/tutorial/toh-pt6.md]: results, as follows.

File: aio/content/tutorial/toh-pt6.md [hunk 6]
```diff
@@ -413,143 +369,125 @@ In this last exercise, you learn to chain `Observable` operators together
...
-As expected, the `*ngFor` repeats hero objects.
+### `AsyncPipe`
 
-Look closely and you'll see that the `*ngFor` iterates over a list called `heroes$`, not `heroes`.
+The `*ngFor` repeats hero objects. Notice that the `*ngFor` iterates over a list called `heroes$`, not `heroes`.
 
-<code-example path="toh-pt6/src/app/hero-search/hero-search.component.html" region="async"></code-example>
+<code-example path="toh-pt6/src/app/hero-search/hero-search.component.html" header="src/app/hero-search/hero-search.component.html" region="async"></code-example>
 
 The `$` is a convention that indicates `heroes$` is an `Observable`, not an array.
```
Comment (jbogarthyde) [inline aio/content/tutorial/toh-pt6.md]: move this sentence up before example?

File: aio/content/tutorial/toh-pt6.md [hunk 7]
```diff
@@ -413,143 +369,125 @@ In this last exercise, you learn to chain `Observable` operators together
...
 
-The `*ngFor` can't do anything with an `Observable`.
-But there's also a pipe character (`|`) followed by `async`,
-which identifies Angular's `AsyncPipe`.
-
-The `AsyncPipe` subscribes to an `Observable` automatically so you won't have to
+Since `*ngFor` can't do anything with an `Observable`, use the
+pipe character (`|`) followed by `async`, which identifies Angular's `AsyncPipe` and subscribes to an `Observable` automatically so you won't have to
```
Comment (jbogarthyde) [inline aio/content/tutorial/toh-pt6.md]: ...followed by `async`. This identifies...

**Review by jbogarthyde — COMMENTED** [review]
Copy edit suggestions

File: aio/content/tutorial/toh-pt2.md [hunk 8]
```diff
@@ -47,18 +45,18 @@ Make it look like this:
...
 The [`*ngFor`](guide/template-syntax#ngFor) is Angular's _repeater_ directive.
 It repeats the host element for each element in a list.
 
-In this example
+The syntax in this example is as follows:
 
-* `<li>` is the host element
-* `heroes` is the list from the `HeroesComponent` class.
+* `<li>` is the host element.
+* `heroes` is the list from the `HeroesComponent` class, the mock heroes list.
```
[the remark this answers is not in the document]
Comment (kapunahelewong) [reply aio/content/tutorial/toh-pt2.md]: Fixed!

File: aio/content/tutorial/toh-pt2.md [hunk 9]
```diff
@@ -127,9 +125,8 @@ This is an example of Angular's [event binding](guide/template-syntax#event-bind
 The parentheses around `click` tell Angular to listen for the `<li>` element's  `click` event.
 When the user clicks in the `<li>`, Angular executes the `onSelect(hero)` expression.
 
-`onSelect()` is a `HeroesComponent` method that you're about to write.
-Angular calls it with the `hero` object displayed in the clicked `<li>`,
-the same `hero` defined previously in the `*ngFor` expression.
+`onSelect()` is a `HeroesComponent` method that Angular will call with the
+`hero` object displayed in the clicked `<li>`&mdash;the same `hero` defined previously in the `*ngFor` expression.
```
[the remark this answers is not in the document]
Comment (kapunahelewong) [reply aio/content/tutorial/toh-pt2.md]: Fixed.

File: aio/content/tutorial/toh-pt4.md [hunk 10]
```diff
@@ -56,36 +56,32 @@ The implementation in _this_ tutorial will continue to deliver _mock heroes_.
...
-as you will do [below](#inject). You do this by registering a _provider_. A provider is something that can create or deliver a service; in this case, it instantiates the `HeroService` class to provide the service.
-
-Now, you need to make sure that the `HeroService` is registered as the provider of this service. 
-You are registering it with an _injector_, which is the object that is responsible for choosing and injecting the provider where it is required. 
+You must make the `HeroService` available to the dependency injection system
+before Angular can _inject_ it into the `HeroesComponent` by registering a _provider_. A provider is something that can create or deliver a service; in this case, it instantiates the `HeroService` class to provide the service.
 
-By default, the Angular CLI command `ng generate service` registers a provider with the _root injector_ for your service by including provider metadata in the `@Injectable` decorator. 
+To make sure that the `HeroService` is registered as the provider of this service, register it with the _injector_, which is the object that is responsible for choosing and injecting the provider where the app requires it.
```
[the remark this answers is not in the document]
Comment (kapunahelewong) [reply aio/content/tutorial/toh-pt4.md]: Fixed.

File: aio/content/tutorial/toh-pt2.md [hunk 11]
```diff
@@ -142,10 +139,11 @@ to the component's `selectedHero`.
...
 </code-example>
 
-### Update the details template
+### Add a details section
 
-The template still refers to the component's old `hero` property which no longer exists.
-Rename `hero` to `selectedHero`.
+Currently, you have a list in the component template. To click on a hero on the list
+and reveal details about that hero, you need a section for the details to render in the
+component template. Add the following to `heroes.component.html` beneath the list section:
```
[the remark this answers is not in the document]
Comment (kapunahelewong) [reply aio/content/tutorial/toh-pt2.md]: Done.

File: aio/content/tutorial/toh-pt5.md [hunk 12]
```diff
@@ -36,79 +36,74 @@ Use the CLI to generate it.
...
+Since `AppRoutingModule` already imports `HeroesComponent`, you can use it in the `routes` array:
+
+<code-example path="toh-pt5/src/app/app-routing.module.ts" header="src/app/app-routing.module.ts"
+  region="heroes-route">
+</code-example>
+
 A typical Angular `Route` has two properties:
 
 1. `path`: a string that matches the URL in the browser address bar.
 1. `component`: the component that the router should create when navigating to this route.
```
[the remark this answers is not in the document]
Comment (kapunahelewong) [reply aio/content/tutorial/toh-pt5.md]: Done.

File: aio/content/tutorial/toh-pt6.md (hunk 4 above)
Comment (kapunahelewong) [reply aio/content/tutorial/toh-pt6.md]: Done.

File: aio/content/tutorial/toh-pt6.md (hunk 3 above)
Comment (kapunahelewong) [reply aio/content/tutorial/toh-pt6.md]: Done.

File: aio/content/tutorial/toh-pt6.md (hunk 5 above)
Comment (kapunahelewong) [reply aio/content/tutorial/toh-pt6.md]: Done.

File: aio/content/tutorial/toh-pt6.md (hunk 6 above)
Comment (kapunahelewong) [reply aio/content/tutorial/toh-pt6.md]: Done.

File: aio/content/tutorial/toh-pt6.md (hunk 7 above)
Comment (kapunahelewong) [reply aio/content/tutorial/toh-pt6.md]: Done!

**Review by IgorMinar — APPROVED** [review]

**Pull request closed without merging**